modelci/cli/modelhub.py

```diff
@@ -39,11 +48,98 @@ def register_model(ymal_path):
     Args:
         ymal_path ([type]): a ymal file that contains model registeration info
     """
-    
+
     register_model_from_yaml(ymal_path)
     logger.info("model published")
 
 
+def _publish_model_from_requests(model_in_yaml: MLModelInYaml):
+    payload = {'convert': model_in_yaml.convert, 'profile': model_in_yaml.profile}
+    data = model_in_yaml.dict(use_enum_values=True, exclude_none=True, exclude={'convert', 'profile', 'weight'})
+    form_data = {k: str(v) for k, v in data.items()}
+    file_or_dir = model_in_yaml.weight
+
+    files = list()
+    key = 'files'
+    try:
+        # read weight file
+        if file_or_dir.is_dir():
+            for file in filter(Path.is_file, file_or_dir.rglob('*')):
+                name = Path(file).relative_to(file_or_dir.parent)
+                files.append((key, (str(name), open(file, 'rb'), 'application/example')))
+        else:
+            files.append((key, (file_or_dir.name, open(file_or_dir, 'rb'), 'application/example')))
+        with requests.post(
+                f'http://{SERVER_HOST}:{SERVER_PORT}/api/v1/model/',
+                params=payload, data=form_data, files=files,
+        ) as r:
+            typer.echo(r.json(), color=True)
+    finally:
+        for file in files:
+            file[1][1].close()
+
+
+@app.command()
```

Comment:
publish means publish from ymal

check my comments above.

Please now come up with something like 'modelhub_typer' or 'publish-ymal'

Reply:
if you need test the ’publish with many input‘ function you may try to add a new "publish_input" function and remove them later.

Also it is better to move your curl test command to a unit test file

Reply:
1. Merge two publish CLI functions. Upload with typed meta data: `modelci modelhub publish --xxx --xxx --xxx "xxxx" --xxx "xxxx"`
2. modehub_typer is a typer version of this command group. Since there are other CLI function implemented using `click`, set typer and click application to the same sub command name (modelhub) many incur errors.
Temporary, the typer application will using `modelci modelhub_type`.

Reply:
Noted for test scripts

Reply:
> 1. Merge two publish CLI functions. Upload with typed meta data: `modelci modelhub publish --xxx --xxx --xxx "xxxx" --xxx "xxxx"`
> 
> 2. modehub_typer is a typer version of this command group. Since there are other CLI function implemented using `click`, set typer and click application to the same sub command name (modelhub) many incur errors.
> 
> Temporary, the typer application will using `modelci modelhub_type`.

1. Good 
2. I understand the reason you did this. However, this will make users fail to use the system. No matter they check wiki or tutorial, they do not know how to register a model 

We shall not add or change CLI too frequently otherwise both old users or beginners can not join our development or start the system.